svt-av1: 2.1.2 -> 2.2.0

hexacera opened the pull request:
## Description of changes

As a follow-up of #228556, this PR enables some additional optimization flags:

- `-DSVT_AV1_PGO=ON`: Enables profile-guided optimization, according to [official build instructions](https://gitlab.com/AOMediaCodec/SVT-AV1/-/blob/master/Docs/Build-Guide.md#linux-operating-systems-64-bit): "(LTO and PGO can be combined, for maximum gains)". See [build.sh](https://gitlab.com/AOMediaCodec/SVT-AV1/-/blob/master/Build/linux/build.sh?ref_type=heads).
- `-DCMAKE_BUILD_TYPE=Release`, `-DNATIVE=ON` are also from [the source that #228556 referenced](https://old.reddit.com/r/AV1/comments/z9fo51/svtav1_on_arm/iygjqfx/).

Enabling PGO only works with 2.2.0, so this PR also bumps version from 2.1.2 to 2.2.0, see [release notes](https://gitlab.com/AOMediaCodec/SVT-AV1/-/tags/v2.2.0).

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

hexacera committed:
svt-av1: 2.1.2 -> 2.2.0

Release notes: https://gitlab.com/AOMediaCodec/SVT-AV1/-/tags/v2.2.0

emilazy commented:
Note that PGO is in conflict with reproducibility, and this package is in the closure of several critical packages like `ffmpeg`, so this would most likely compromise the reproducibility of the installer ISOs.

jopejoe1 commented on pkgs/tools/video/svt-av1/default.nix:
```diff
@@ -26,7 +26,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DNATIVE=ON"
```
This would also breake reproducibility as it sets march to native.

jopejoe1 commented on pkgs/tools/video/svt-av1/default.nix:
```diff
@@ -26,7 +26,10 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
```
This is already set to Release by default

jopejoe1 reviewed (changes requested):
I would recommend droping the second commit as it currently breaks reproducibility.

hexacera commented:
Thanks @emilazy and @jopejoe1! I see that's why these flags aren't enabled already. I have dropped the second commit and updated the PR title such that this PR is a version bump only.

jopejoe1 reviewed (changes requested):
Thanks for updating the pr, the changes now look good to me, and I tested that they work by encoding a video in FFmpeg with it and playing the resulting video.

Just one issue, that I overlooked in my previous review is that this should be targeting staging, could you please rebase the changes onto it, while follow the instructions from https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging carefully.
Thanks!

hexacera commented:
Thanks @jopejoe1 ! I have switched target to staging.

jopejoe1 reviewed (approved):
Thanks!
